docs/okeps/okep-5494-ovn-kubernetes-mcp-server.md

```diff
@@ -0,0 +1,740 @@
+│  └──────────────────────────────────────────────────────────────────────┘    │
+└──────────────────────────────────────────────────────────────────────────────┘
+                                       │
+                           ┌───────────┼───────────┐
+                           │           │           │
+                           ▼           ▼           ▼
+                      ┌────────┐  ┌────────┐  ┌────────┐
+                      │ Node1  │  │ Node2  │  │ NodeN  │
+                      │        │  │        │  │        │
+                      │ovnkube-│  │ovnkube-│  │ovnkube-│
+                      │node pod│  │node pod│  │node pod│
+                      │        │  │        │  │        │
+                      │ ovn-nb │  │ ovn-nb │  │ ovn-nb │
+                      │ ovn-sb │  │ ovn-sb │  │ ovn-sb │
+                      │ ovs    │  │ ovs    │  │ ovs    │
+                      └────────┘  └────────┘  └────────┘
+
+Data Flow Examples:
+├─ Live: kubectl exec ovnkube-node-xyz -c nb-ovsdb -- ovn-nbctl show
+├─ Live: oc debug node/worker-1 -- ovs-ofctl dump-flows br-int
+├─ Live: oc debug node/worker-1 -- ip route show table all
+├─ Offline: Parse must-gather/ovn-kubernetes/ovn-northbound.db
+├─ Offline: Parse sos-report/node-1/openvswitch/ovs-ofctl_dump-flows
+└─ Offline: Parse sos-report/node-1/networking/ip_route
+```
+
+## Implementation Details of the OVN-Kubernetes MCP Server
+
+See the Alternatives section for other ideas that were discarded.
+
+### Chosen Approach: Direct CLI Tool Exposure (Idea1)
+
+The initial implementation takes a pragmatic approach by directly exposing the
+existing CLI tools (`ovn-nbctl`, `ovn-sbctl`, `ovs-vsctl`, etc.) as MCP tools.
+While this approach may seem less elegant than creating higher-level wrapper
+abstractions to connect to the database (see discarded alternatives), it
+offers the fastest path to value for OVN-Kubernetes engineers who are already
+familiar with these tools but want to leverage LLM assistance for command
+construction and output analysis. The CLI-based approach is also optimal for
+exposing the complete troubleshooting toolkit across all layers of the stack.
+Most other discarded ideas only addressed OVSDB access and reusability
+concerns while failing to provide the holistic set of tools needed for
+comprehensive root cause analysis like running packet traces.
+
+The MCP server acts as a secure execution bridge, translating natural language
+troubleshooting requests into appropriate CLI commands.
+
+**Advantages**:
+
+* **Fastest Time to Value**: Leverages existing tools that engineers already
+  know
+* **Zero Deployment Overhead**: All required CLI binaries are already present
+  in the pods and nodes within the cluster
+* **Comprehensive Coverage**: Only approach that provides access to the
+  complete troubleshooting toolkit across all stack layers
+* **Security Controls**: Enables proper RBAC and read-only access enforcement
```

Comment:
It is not clear to me how RBAC is achieved here.

Reply:
me neither, :) so its a great casual slip that needs more expansion.

I feel like I forgot what you, me and @dceara discussed. Did we say its not possible to restrict access using certs for ctl commands?

Reply:
I've rephrased this to remove RBAC for now and go with read-only since as of today we have no RBAC for CLI commands.